terra/compute/singularity.py

```diff
@@ -0,0 +1,92 @@
+import os
+from subprocess import PIPE
+
+from terra.compute.base import BaseCompute
+from terra.compute.container import ContainerService
+from terra.compute.utils import just
+from terra.compute.base import ServiceRunFailed
+
+from terra.logger import getLogger
+logger = getLogger(__name__)
+
+
+class Compute(BaseCompute):
+  def run_service(self, service_info):
+    '''
+    Use the service class information to run the service runner in a docker
+    using
+
+    .. code-block:: bash
+
+        just --wrap singular-compose \\
+            run {service_info.compose_service_name} \\
+            {service_info.command}
+    '''
+    pid = just("singular-compose",
+               'run', service_info.compose_service_name,
+               *(service_info.command),
+               env=service_info.env)
+
+    if pid.wait() != 0:
+      raise ServiceRunFailed()
+
+  def config_service(self, service_info, extra_compose_files=[]):
+    '''
+    Returns the ``singular-compose config-null`` output
+    '''
+
+    args = ["singular-compose",
+            '-f', service_info.compose_file] + \
+        sum([['-f', extra] for extra in extra_compose_files], []) + \
+        ['config-null', service_info.compose_service_name]
+
+    pid = just(*args, stdout=PIPE,
+               env=service_info.env)
+    data = pid.communicate()[0]
+
+    data = data.split(b'\0^')
+    data = dict(zip([header.decode() for header in data[::2]],
+                    [[chunk.decode() for chunk in group.split(b'\0.')]
+                     for group in data[1::2]]))
```

Comment:
Very clean, but maybe you could add a comment explaining the data format is "[header] \0. [chunk] \0. [chunk] \0^ [header]..."